service/app/controllers/notesmanagement/NotesController.java

```diff
@@ -68,17 +55,17 @@
     try {
       JsonNode requestData = request().body().asJson();
       ProjectLogger.log("Update note request: " + requestData, LoggerEnum.INFO.name());
-      Request reqObj = (Request) mapper.RequestMapper.mapRequest(requestData, Request.class);
-      RequestValidator.validateNoteId(noteId);
-      reqObj.setOperation(ActorOperations.UPDATE_NOTE.getValue());
-      reqObj.setRequestId(ExecutionContext.getRequestId());
-      reqObj.setEnv(getEnvironment());
-      HashMap<String, Object> innerMap = new HashMap<>();
+      Request request = createAndInitRequest(ActorOperations.UPDATE_NOTE.getValue(), requestData);
```

Comment:
Can you simplify this further by adding a new method (for instance, handlePostRequest) in BaseController?

public Promise updateNote() {
    handlePostRequest(noteId, ActorOperations.UPDATE_NOTE.getValue(), requestData, (request)-> new NoteRequestValidator().validateUpdateNoteRequest(request));
}